core/trino-main/src/main/java/io/trino/execution/scheduler/PipelinedQueryScheduler.java

```diff
@@ -1082,7 +1083,8 @@ public void stateChanged(QueryState newState)
                         writerTasksProvider,
                         nodeScheduler.createNodeSelector(session, Optional.empty()),
                         executor,
-                        getWriterMinSize(session));
+                        getWriterMinSize(session),
```

Comment:
When `partitionCount` can be empty? I think it's always set in `AddExchanges`.

I don't think you can use `getMaxWriterTasksCount(session)` because you would ignore `getMaxWriterTasks`